pkg/controller/kubefed/kubefed_controller.go

```diff
@@ -150,22 +150,24 @@ func resourceEnvUpdate(scope kubefedv1alpha1.InstallationScope, ns, name string)
 		case "deployment":
 			if containers, ok, err := unstructured.NestedSlice(u.Object,
 				"spec", "template", "spec", "containers"); ok {
-				if envs, envOk, envErr := unstructured.NestedSlice(containers[0].(map[string]interface{}), "env"); envOk {
-					if !checkEnvExists(envs, "name", "DEFAULT_KUBEFED_SCOPE") {
-						fse := map[string]interface{}{"name": "DEFAULT_KUBEFED_SCOPE", "value": kubefedScope}
-						envs = append(envs, fse)
+				envs, envOk, envErr := unstructured.NestedSlice(containers[0].(map[string]interface{}), "env")
```

Comment:
If env slice doesn't exist on the containers[0], then the return value in envs will be nil. So before calling Line 155, do you need to make() the slice albeit empty?